src/solvers/flattening/bv_utils.cpp

```diff
@@ -482,26 +482,49 @@ bvt bv_utilst::shift(const bvt &src, const shiftt s, std::size_t dist)
   bvt result;
   result.resize(src.size());
 
+  // 'dist' is user-controlled, and thus arbitary.
+  // We thus must guard against the case in which i+dist overflows.
+  // We do so by considering the case dist>=src.size().
+
   for(std::size_t i=0; i<src.size(); i++)
   {
     literalt l;
 
     switch(s)
     {
-    case shiftt::LEFT:
-      l=(dist<=i?src[i-dist]:const_literal(false));
+    case shiftt::SHIFT_LEFT:
+      if(dist>=src.size())
+        l=const_literal(false);
+      else
+        l=(dist<=i?src[i-dist]:const_literal(false));
+      break;
+
+    case shiftt::SHIFT_ARIGHT:
+      if(dist>=src.size())
+        l=src[src.size()-1];
+      else
```

Comment:
Largely redundant with the else case below; the case that isn't covered is arithmetic overflow in `i+dist`, but that wouldn't be covered by the above either.

Reply:
Now covered.